pkg/util/spanlogger/spanlogger.go

```diff
@@ -31,18 +44,29 @@ func New(ctx context.Context, method string, kvps ...interface{}) (*SpanLogger,
 	return logger, ctx
 }
 
-// FromContext returns a span logger using the current parent span.
-// If there is no parent span, the Spanlogger will only log to global logger.
+// FromContext returns a span logger using the current parent span. If there
+// is no parent span, the SpanLogger will only log to the logger
+// in the context. If the context doesn't have a logger, the global logger
+// is used.
 func FromContext(ctx context.Context) *SpanLogger {
+	return FromContextWithFallback(ctx, util.Logger)
+}
+
+// FromContextWithContext returns a span logger using the current parent span.
+// IF there is no parent span, the SpanLogger will only log to the logger
+// within the context. If the context doesn't have a logger, the fallback
+// logger is used.
+func FromContextWithFallback(ctx context.Context, fallback log.Logger) *SpanLogger {
+	logger, ok := ctx.Value(loggerCtxKey).(log.Logger)
```

Comment:
I don't see any code that would be *storing* value under this key. What am I missing?

Reply:
Thanks, I copied this in a rush from [my temporary fork of the package](https://github.com/rfratto/agent/blob/b2af97528144f62fbab936ef2fedeae212e2e6a7/pkg/util/spanlogger/spanlogger.go#L40) and forgot that one part 😞 

I'll add a test!